src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -640,6 +649,15 @@ private SafeCurlSlistHandle SetRequestHeaders(SafeCurlHandle handle, HttpRequest
                     rawHandle = Interop.libcurl.curl_slist_append(rawHandle, header);
                     retVal.SetHandle(rawHandle);
                 }
+
+                // Since libcurl always adds a Transfer-Encoding header, we need to explicitly block
+                // it if caller specifically does not want to set the header
+                if (request.Headers.TransferEncodingChunked.HasValue && !request.Headers.TransferEncodingChunked.Value)
```

Comment:
I think that this one has to remain as-is: the condition is only true when `TransferEncodingChunked` has a value and that value is `false`. `GetValueOrDefault()